core/nwb.base.yaml

```diff
@@ -234,3 +275,27 @@ groups:
   - neurodata_type_inc: Image
     doc: Images stored in this collection.
     quantity: '+'
+  - name: order
+    neurodata_type_inc: VectorData
+    dims:
+    - num_images
+    shape:
+    - null
+    dtype:
+      target_type: Image
+      reftype: object
+    doc: Ordered dataset of references to Image objects stored in the parent group.
+    quantity: '?'
```

Comment:
Does this work in combination with the ImageStacks type. Can you select images with this both from individual images and a stack of images?

Reply:
No, but does it need to?

I'm not sure why one would want an ordered collection of ImageStack objects and I would rather not add things that have not yet been asked for or have a clear use case.

Reply:
I think I may be misunderstanding the intended use of ImageStack. Can you clarify what the intended use of ``ImageStack`` vs a collection of ``Image`` objects is?

Reply:
The primary use case is when a user wants to store the times associated with the presentation of a series of stimulus images. This is done in an `IndexSeries`. `IndexSeries/data` contains a list of indices to images stored in either an `ImageStack` or an `Images` object that has `Images/order` set.

An `ImageStack` is to be used when a user wants to store multiple images with the same size. It is similar to a TIFF stack. I think it is an intuitive representation of these data. Storing images in `Images` with the `Images/order` dataset covers a broader use case: when a user wants to store multiple images that do not necessarily have the same size or number of channels in some order. 

We could remove the `ImageStack` and `ImageStacks` types so that there is only one way to store an ordered collection of Images. That seems simpler, and after further discussion, I think @bendichter 
is in favor of this approach. I am now too. We can always add `ImageStack` and `ImageStacks` later on if it is needed.

Doing so would mean that given an `IndexSeries`, to find out which image was presented at the first timestamp, the user (or API) would have to look do:
```python
image_index = index_series.data[0]
image = index_series.indexed_images.order[image_index]
```
This is a little indirect but can be explained.

Also I think `order` should be renamed `ordered_images`.

Reply:
Yeah, I think `Images` should have an optional `ordered_images` dataset that stores references to the images in that container in order. That will designate the order of the corresponding images so `IndexSeries` can point to the `Images` object and we know which image `IndexSeries` is referring to with each element value